src/net.cpp

```diff
@@ -653,11 +666,19 @@ int V1TransportDeserializer::readHeader(const char *pch, unsigned int nBytes)
         hdrbuf >> hdr;
     }
     catch (const std::exception&) {
+        LogPrint(BCLog::NET, "HEADER ERROR - UNABLE TO DESERIALIZE\n");
+        return -1;
+    }
+
+    // Check start string, network magic
+    if (memcmp(hdr.pchMessageStart, chain_params.MessageStart(), CMessageHeader::MESSAGE_START_SIZE) != 0) {
+        LogPrint(BCLog::NET, "HEADER ERROR - MESSAGESTART (%s, %u bytes), got %s\n", hdr.GetCommand(), hdr.nMessageSize + CMessageHeader::HEADER_SIZE, hdr.GetMessageStart());
```

Comment:
The command and size outputs are being kept to stay consistent with other error messages.  Both the codebase and documentation aren't consistent between the terms "Message Start String" and "Network Magic", I'm preferring "Message Start String" here as I think it's the older one, and neither are sufficiently descriptive out of context anyways.

Reply:
I don't think it's necessary to print out the expected MessageStart in the debug message.  The MessageStart is set when bitcoind starts up and doesn't change after that.  No one should be confused as to what the expected value is.  Additionally I don't print out the message's MessageStart here, because I can't find a situation where it would be relevant.  I'm open to changing my stance on that if someone can propose when that would be useful.